packages/app/src/cli/services/function/ui/components/Replay.tsx

```diff
@@ -0,0 +1,168 @@
+import {setupExtensionWatcherForReplay} from './hooks/setup-extension-watcher-for-replay.js'
+import {FunctionRunData} from '../../replay.js'
+import {ExtensionInstance} from '../../../../models/extensions/extension-instance.js'
+import {FunctionConfigType} from '../../../../models/extensions/specifications/function.js'
+import {AppInterface} from '../../../../models/app/app.js'
+import {prettyPrintJsonIfPossible} from '../../../app-logs/utils.js'
+import figures from '@shopify/cli-kit/node/figures'
+import {AbortController} from '@shopify/cli-kit/node/abort'
+import React, {FunctionComponent} from 'react'
+import {Box, Text, Static} from '@shopify/cli-kit/node/ink'
+
+export interface ReplayProps {
+  selectedRun: FunctionRunData
+  abortController: AbortController
+  app: AppInterface
+  extension: ExtensionInstance<FunctionConfigType>
+}
+
+interface FunctionRun {
+  type: 'functionRun'
+  input: string
+  output: string
+  logs: string
+  name: string
+  size: number
+  memory_usage: number
+  instructions: number
+}
+
+interface SystemMessage {
+  type: 'systemMessage'
+  message: string
+}
+
+type ReplayLog = FunctionRun | SystemMessage
+
+const Replay: FunctionComponent<ReplayProps> = ({selectedRun, abortController, app, extension}) => {
+  const {logs, isAborted, canUseShortcuts, statusMessage, recentFunctionRuns, error} = setupExtensionWatcherForReplay({
```

Comment:
`setupExtensionWatcherForReplay` is effectively a React hook. The convention is to name is `useXYZ`. So in this case, I'd name it something like `useFunctionWatcher` (since it's also specific to Functions).